### PR TITLE
Feature/97 matchmaking agent api

### PR DESCRIPTION
**Pull request by Martin0815bla**
## WHAT

This container separates the Matchmaking Agent's functionality from the Agent Plane Container. As a result, the RDF Store is disjoined from the public-facing Agent Plane container.

## WHY

 It caters for the need of small and medium sized companies who want to directly upload/provide use case data as well as for security-aware companies who want to separate data storage from public API even for meta-data. This container also has the ability to be used by the Simple Graph Exchanger Feature, adding the ability to host business data from the RDF Store over the Matchmaking Agent Communication between the Agent Plane Container and the Matchmaking Agent by a REST API on the docker (internal) network.

## FURTHER NOTES

To make use of the matchmaking agent set the right dataplane.properties settings in tx-knowledge-agents-edc (agent-plane-protocol)

Closes # 
Feature97

### CODE REVIEW

File: charts/matchmaking-agent/.helmignore
Comment (drcgjung) [inline charts/matchmaking-agent/.helmignore]: need to adapt copyright period.

**Review by drcgjung — APPROVED** [review]
This is mostly a migration/carveout of code from https://github.com/eclipse-tractusx/knowledge-agents-edc/tree/main/agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc into a new servlet container (glassfish) together with some EDC client stubs (to render it independent of EDC).
It represents a new product/chart which provides external agent functionality (SPARQL engine, federated catalogue) independent of EDC (data plane) such that it can be used to host business data as well without being exposed to the dataspace/public internet.
As such it represents a big advantage for security architecture.
Copyright attachments and dependencies have been rechecked, such that no fresh IP check needs to made.

see also https://github.com/eclipse-tractusx/knowledge-agents-edc/pull/184

File: matchmaking/src/main/docker/Dockerfile
Comment (almadigabor) [inline matchmaking/src/main/docker/Dockerfile:28]: Based on this [TRG](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-02#use-base-image-as-is) it is discouraged to modify a base image like this. It would make the image scan and package list provided by DockerHub unusable.

**Comment by almadigabor** [comment]
Hey, I would also suggest implementing the dependencies workflow to have an automated check on each PR like [here](https://github.com/eclipse-tractusx/portal-backend/blob/main/.github/workflows/dependencies.yaml). This would make sure that we don't merge PRs that have outdated DEPENDENCIES files. Also can you clarify here how the dependencies work in this repo? Are they consolidated into a single file?

**Comment by almadigabor** [comment]
> This is mostly a migration/carveout of code from https://github.com/eclipse-tractusx/knowledge-agents-edc/tree/main/agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc into a new servlet container (glassfish) together with some EDC client stubs (to render it independent of EDC). It represents a new product/chart which provides external agent functionality (SPARQL engine, federated catalogue) independent of EDC (data plane) such that it can be used to host business data as well without being exposed to the dataspace/public internet. As such it represents a big advantage for security architecture. Copyright attachments and dependencies have been rechecked, such that no fresh IP check needs to made.
> 
> see also [eclipse-tractusx/knowledge-agents-edc#184](https://github.com/eclipse-tractusx/knowledge-agents-edc/pull/184)

Hi @drcgjung, even though it is a migration, an IP check is necessary as the contribution is coming from catenax-ng and the number of line changes are way above 1000. See [TRG7.03](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-03#checking-contributions-by-a-eclipse-contributor) for more info.

**Comment by Siegfriedk** [comment]
> This is mostly a migration/carveout of code from https://github.com/eclipse-tractusx/knowledge-agents-edc/tree/main/agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc into a new servlet container (glassfish) together with some EDC client stubs (to render it independent of EDC). It represents a new product/chart which provides external agent functionality (SPARQL engine, federated catalogue) independent of EDC (data plane) such that it can be used to host business data as well without being exposed to the dataspace/public internet. As such it represents a big advantage for security architecture. Copyright attachments and dependencies have been rechecked, such that no fresh IP check needs to made.
> 
> see also [eclipse-tractusx/knowledge-agents-edc#184](https://github.com/eclipse-tractusx/knowledge-agents-edc/pull/184)

I would agree if you would have had smaller and more focused PRs doing clearly only one thing and not multiply things. Or if you would have developed this IN eclipse-tractusx (a branch) and not from an outside repository.

This definitly needs an IP check and proper procedure

**Comment by almadigabor** [comment]
I've opened the IP ticket, you can it's track the progress [here](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14224).

File: matchmaking/src/main/docker/Dockerfile
Comment (drcgjung) [reply matchmaking/src/main/docker/Dockerfile:28]: will be resolved.

Comment (drcgjung) [reply matchmaking/src/main/docker/Dockerfile:28]: will be resolved

Comment (drcgjung) [reply matchmaking/src/main/docker/Dockerfile:28]: will be resolved. Thx for the heads up. Martin used curl instead of wget for testing on his machine and commited the change unfortunately.

**Comment by drcgjung** [comment]
> Hey, I would also suggest implementing the dependencies workflow to have an automated check on each PR like [here](https://github.com/eclipse-tractusx/portal-backend/blob/main/.github/workflows/dependencies.yaml). This would make sure that we don't merge PRs that have outdated DEPENDENCIES files. Also can you clarify here how the dependencies work in this repo? Are they consolidated into a single file?

will do

**Comment by drcgjung** [comment]
> > This is mostly a migration/carveout of code from https://github.com/eclipse-tractusx/knowledge-agents-edc/tree/main/agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc into a new servlet container (glassfish) together with some EDC client stubs (to render it independent of EDC). It represents a new product/chart which provides external agent functionality (SPARQL engine, federated catalogue) independent of EDC (data plane) such that it can be used to host business data as well without being exposed to the dataspace/public internet. As such it represents a big advantage for security architecture. Copyright attachments and dependencies have been rechecked, such that no fresh IP check needs to made.
> > see also [eclipse-tractusx/knowledge-agents-edc#184](https://github.com/eclipse-tractusx/knowledge-agents-edc/pull/184)
> 
> Hi @drcgjung, even though it is a migration, an IP check is necessary as the contribution is coming from catenax-ng and the number of line changes are way above 1000. See [TRG7.03](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-03#checking-contributions-by-a-eclipse-contributor) for more info.

I though so. Yes, lets do an IP check on the patch text.

**Review by drcgjung — APPROVED** [review]
- IP check approved after fixing an erroneous SPDX line
- unify setup-java steps.

**Review by almadigabor — APPROVED** [review]

**Pull request merged**
